main_script.py

```diff
@@ -0,0 +1,348 @@
+        records: records from fastq parced by SeqIO
+        length_bound: interval for the of acceptable sequense length in number of nucleotide
+    :return:(dict) new dictionary consists of selected sequences
+    """
+    new_records = []
+    for record in records:
+        if (length_bounds_both_side[1] >= len(record.seq) >= length_bounds_both_side[0]):
+            new_records.append(record)
+    print(new_records)
+    return new_records
+
+
+def filter_quality(records, quality_threshold=0) -> list:
+    """
+    This function selects  FASTQ sequences with appropriate average nucleotide read quality
+    parameters:
+        seqs: dictionary of FASTQ sequences {name: (sequence, quality)}
+        quality_threshold: threshold value for average quality per nucleotide (phred33 scale)
+    :return:(dict) records for selected sequences
+    """
+    new_records = []
+    for record in records:
+        if (sum(record.letter_annotations["phred_quality"])/len(record.seq) >= quality_threshold):
+            new_records.append(record)
+    print(new_records)
+    return new_records
+
+
+def fastq_filtration(input_fastq, gc_bounds=(0, 100), length_bounds=(0, 2 ** 32), quality_threshold=0, output_fastq=''):
+    """
+    This function provides you the opportunity to filter the FASTQ file to select sequences
+    that fit  requirements on 5 parameters: input and output(optional) files, length, GC composition,
+    and quality of the reed
+    :parameters
+        input_fastq: path to fastq file
+        gc_bounds: (tuple) interval for the of acceptable GC content, in %
+        length_bounds: (tuple) interval for the of acceptable sequence length in number of nucleotide
+        quality_threshold: (float) threshold value for average quality per nucleotide (phred33 scale)
+        output_fastq = name of output file, ./fastq_filtrator_resuls/output_fastq, if it is not defined,
+        it will be the same of the input file
+
+    """
+    if not os.path.isdir('fastq_filtrator_resuls'):
+        os.mkdir('fastq_filtrator_resuls')
+    if output_fastq == '':
+        output_fastq = os.path.join('fastq_filtrator_resuls', os.path.basename(input_fastq))
+    else:
+        output_fastq = os.path.join('fastq_filtrator_resuls', output_fastq + ".fasta")
+    if type(gc_bounds) == float or type(gc_bounds) == int:
+        gc_bounds_both_side = (0, gc_bounds)
+    else:
+        gc_bounds_both_side = gc_bounds
+    if type(length_bounds) == int:
+        length_bounds_both_side = (0, length_bounds)
+    else:
+        length_bounds_both_side = length_bounds
+    records = list(SeqIO.parse(input_fastq, "fastq"))
+    filtered_records_l = filter_length(records, length_bounds_both_side)
+    filtered_records_gc = filter_gc(filtered_records_l, gc_bounds_both_side)
+    filtered_records_q = filter_quality(filtered_records_gc, quality_threshold)
+    SeqIO.write((record for record in filtered_records_q), output_fastq, "fastq")
+    return
+    
+
+
+

```

Comment:
Многовато пустных строк, да и return не нужен:)
```suggestion
    records = list(SeqIO.parse(input_fastq, "fastq"))
    filtered_records_l = filter_length(records, length_bounds_both_side)
    filtered_records_gc = filter_gc(filtered_records_l, gc_bounds_both_side)
    filtered_records_q = filter_quality(filtered_records_gc, quality_threshold)
    SeqIO.write((record for record in filtered_records_q), output_fastq, "fastq")
    
```